common/handy/set.go

```diff
@@ -0,0 +1,26 @@
+package handy
+
+// Set data structure that can hold any type of data.
+type Set[T comparable] map[T]bool
```

Comment:
This is slightly cleaner since you're not really doing anything with the booleans.

```suggestion
type Set[T comparable] map[T]struct{}
```

It's also easier on the memory, in theory.